src/test/java/io/xstefank/wildfly/bot/utils/JsonHandler.java

```diff
@@ -0,0 +1,141 @@
+package io.xstefank.wildfly.bot.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Class responsible for handling and changing properties of the JSON template file.
+ */
+public class JsonHandler {
+
+    private static String PULL_REQUEST = "pull_request";
+    private static String COMMITS = "commits";
+    private static String BODY = "body";
+    private static String TITLE = "title";
+    private static String SHA = "sha";
+    private static String HEAD = "head";
+    private static String ACTION = "action";
+
+    private static JsonNode file;
+
+    private JsonHandler(JsonHandlerBuilder jsonHandlerBuilder) {
+        this.file = jsonHandlerBuilder.jsonFile;
+    }
+
+    /**
+     * Returns the SHA of the first commit in the GitHub PR payload.
+     *
+     * @return the SHA of the first commit in the GitHub PR payload.
+     */
+    public String getCommitSHA() {
+        return file.get(PULL_REQUEST).get(HEAD).get(SHA).textValue();
+    }
+
+    /**
+     * Returns the JSON file using the {@link JsonNode#toPrettyString()}.
+     * The {@code toString} method returns the file as a single line String. The {@code toPrettyString} returns the
+     * file in easily readable way.
```

Comment:
You don't need to leak implementation details, what if you would like to change the Json processing implementation in the future?